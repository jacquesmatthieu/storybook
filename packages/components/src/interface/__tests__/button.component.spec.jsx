import React from 'react';
import { shallow } from 'enzyme';

import Button from '../button.component';

describe('Button', () => {
  let props;

  beforeEach(() => {
    props = {

    };
  });

  const getWrapper = () => shallow(<Button {...props}>Text</Button>);

  it('should render properly', () => {
    // When
    const wrapper = getWrapper();

    // Then
    expect(wrapper).toMatchSnapshot();
  });

  it('should render with given style', () => {
    // Given
    props.color = 'secondary';
    props.hollow = true;

    // When
    const wrapper = getWrapper();

    // Then
    expect(wrapper).toMatchSnapshot();
  });

  it('should render link version', () => {
    // Given
    props.href = 'yolo';

    // When
    const wrapper = getWrapper();

    // Then
    expect(wrapper).toMatchSnapshot();
  });
});
