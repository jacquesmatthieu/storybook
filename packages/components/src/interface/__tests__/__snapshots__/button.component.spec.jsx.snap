// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button should render link version 1`] = `
<buttoncomponent__StyledButton
  color="primary"
  hollow={false}
  href="yolo"
>
  <typographycomponent__Typography
    variant="input"
  >
    Text
  </typographycomponent__Typography>
</buttoncomponent__StyledButton>
`;

exports[`Button should render properly 1`] = `
<buttoncomponent__StyledButton
  color="primary"
  hollow={false}
>
  <typographycomponent__Typography
    variant="input"
  >
    Text
  </typographycomponent__Typography>
</buttoncomponent__StyledButton>
`;

exports[`Button should render with given style 1`] = `
<buttoncomponent__StyledButton
  color="secondary"
  hollow={true}
>
  <typographycomponent__Typography
    variant="input"
  >
    Text
  </typographycomponent__Typography>
</buttoncomponent__StyledButton>
`;
